### ISSUE TITLE
Incorrect positioning on mobile using effect='solid'

### ISSUE THREAD

**Issue by levymoreira** [wwayne/react-tooltip#374]
Code:
```
<button data-tip='bla bla bla blbllblb lbalballbl blaba bla bal bal'>'Test'</button>
<ReactTooltip effect='solid' place='bottom' event='click' />
```

The current mobile result: 
![screen shot 2018-04-11 at 15 08 52](https://user-images.githubusercontent.com/588214/38622098-47e1304e-3d9a-11e8-89d5-682b1b76eff2.png)

Expected result:
(Centered tooltip)
![screen shot 2018-04-11 at 15 04 36](https://user-images.githubusercontent.com/588214/38622021-20bf6490-3d9a-11e8-8a62-7dd4828ef423.png)

**Comment by aronhelser** [comment]
Yes, that result is much more desirable. How did you generate the 'expected result'? some style hacking we could use to fix the problem?

Do you have a suggested fix? I'm not set up to develop or test fixes on mobile...

**Comment by levymoreira** [comment]
To fix it I just changed the left property to 0, my suggestion is when you calculating the left value check if it's mobile if it's just don't set it, once that you are setting it directly in the element.style we just can't change it using CSS (actually we can but we need use !import). 
You can test it easily using the Google Chrome mobile device view.

**Comment by archansel** [comment]
Any update on this? I have this issue as well
![screen shot 2018-07-13 at 11 17 44](https://user-images.githubusercontent.com/816817/42672164-7d2582d6-868e-11e8-8c9c-c80a6296833f.png)
Position render correctly after second render; top (first render) bottom (second render). In my case, `top` property got wrong calculation in the first render but correct in second render

**Comment by mxcl** [comment]
The position is always wrong on first appearance however we use it. Since this issue is 2 years old I figure we're better off finding another package.

**Comment by aronhelser** [comment]
Or you could help debug and contribute a fix!

**Comment by mxcl** [comment]
So…. no, I won’t. I am an extremely prolific contributor to open source, but only if it looks like it's a maintained package. Sorry.

**Comment by briangonzalez** [comment]
@mxcl your comment is gross

**Comment by abdulbasit1248** [comment]
This can be achieved using the place prop. Like if you have a question mark on the left side of the screen as I have. Then use place = right similarly left if the question mark is on the right.

`<ReactTooltip
        id={id}
        type="success"
        effect="solid"
        multiline={true}
        border
        textColor="#fff"
        place="right"
        backgroundColor="#042E60"
      />`

![image](https://user-images.githubusercontent.com/43878789/113669444-36fdd900-96cd-11eb-9ce8-6c81ac72bbca.png)

**Comment by smellai** [comment]
I have this issue too. Looks like first render is correct and subsequent ones are wrong. This happens because I have multiple targets triggering tooltips in the same component and they are affecting each other positioning (because of how [getPosition is implemented](https://github.com/wwayne/react-tooltip/blob/master/src/utils/getPosition.js) I think). If I keep only one tooltip the arrow is positioned correctly, but the tooltip itself overflows the viewport as described above.
